### PR TITLE
feat(actions): enforce semantic pr title

### PR DESCRIPTION
**Pull request by kengou**
## Description

This PR is adding a new action to run on every PR to enforce semantic headers, this is a pre-requisite for CHANGELOG generation. Shared action is not yet available, but In-progress

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

- Related Issue # (issue)
- Closes # (issue)
- Fixes # (issue)

> Remove if not applicable

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help
- [ ] Separate ticket for tests # (issue/pr)

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Added to documentation?

- [ ] 📜 README.md
- [ ] 🤝 Documentation pages updated
- [ ] 🙅 no documentation needed
- [ ] (if applicable) generated OpenAPI docs for CRD changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .github/workflows/ci-pr-title.yaml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            message-provider
+            npm
+            oauth
+            supernova
+            template
+            ui
+            utils
+            url-state-provider
+            version
+            ISSUE-\d+
```
Comment (IvoGoman) [inline .github/workflows/ci-pr-title.yaml:60]: This list of scopes needs to be adapted to Greenhouse context.
I suggest we start with adding every CRD name + 'controller', 'webhook', 'greenhousectl' 
And removing Juno scopes such as `communicator, `carbon`, `doop` etc.

Comment (kengou) [reply .github/workflows/ci-pr-title.yaml]: fixed in [e8a19be](https://github.com/cloudoperators/greenhouse/pull/675/commits/e8a19bea9c8253bf60043ac6645ccf67f54a6086)

File: .github/workflows/ci-pr-title.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+          # Configure which scopes are allowed (newline-delimited).
+          # These are regex patterns auto-wrapped in `^ $`.
+          scopes: |
+            cluster-controller
+            organization-controller
+            plugin-controller
+            plugindefinition-controller
+            team-controller
+            teammembership-controller
+            teamsrbac-controller
```
Comment (IvoGoman) [inline .github/workflows/ci-pr-title.yaml:41]: Let's drop the controller and keep just the CRD names. This leaves more characters for the description.

```suggestion
            clusterkubeconfigs
            clusters
            organizations
            plugindefinitions
            pluginpresets
            plugins
            teammemberships
            teamrolebindings
            teamroles
            teams
```

File: .github/workflows/ci-pr-title.yaml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            build
+            config
+            ci
+            core
```
Comment (IvoGoman) [inline .github/workflows/ci-pr-title.yaml:45]: A generic controller for changes to lifecycle or other common parts
```suggestion
            core
            controller
```

File: .github/workflows/ci-pr-title.yaml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+            clusterkubeconfigs
+            clusters
+            organizations
+            plugindefinitions
+            pluginpresets
+            plugins
+            teammemberships
+            teamrolebindings
+            teamroles
+            teams
```
Comment (kengou) [inline .github/workflows/ci-pr-title.yaml:44]: I see that @uwe-mayer is using `service-proxy` in some PR
```suggestion
            teams
            service-proxy
```

File: .github/workflows/ci-pr-title.yaml [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+            config
+            ci
+            core
+            controller
+            deps
+            docs
+            e2e
+            greenhouse
+            greenhousectl          
+            actions
```
Comment (kengou) [inline .github/workflows/ci-pr-title.yaml:55]: Question if we need a specific one for `renovate` and `Makefile` ? or it should fit under `utils` 
```suggestion
            actions
            renovate
            makefile
```

**Review by IvoGoman — APPROVED** [review]

File: .github/workflows/ci-pr-title.yaml [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+            refactor
+            research
+            style
+            test
+          # Configure which scopes are allowed (newline-delimited).
+          # These are regex patterns auto-wrapped in `^ $`.
+          scopes: |
+            clusterkubeconfigs
+            clusters
+            organizations
```
Comment (uwe-mayer) [inline .github/workflows/ci-pr-title.yaml:37]:
```suggestion
            organizations
            idproxy
```

**Pull request merged**